src/GraphQL.Client/GraphQLHttpClient.cs

```diff
@@ -111,6 +116,62 @@ public IObservable<GraphQLResponse<TResponse>> CreateSubscriptionStream<TRespons
 
     #region Private Methods
 
+    private async Task<GraphQLHttpResponse<TResponse>> SendAPQHttpRequestAsync<TResponse>(GraphQLRequest request, CancellationToken cancellationToken = default)
+    {
+        cancellationToken.ThrowIfCancellationRequested();
+
+        var savedQuery = request.Query;
+        bool useAPQ = false;
+
+        if (request.Query != null && _useAPQ && Options.EnableAutomaticPersistedQueries(request))
+        {
+            // https://www.apollographql.com/docs/react/api/link/persisted-queries/
+            const int APQ_SUPPORTED_VERSION = 1;
+            useAPQ = true;
+            // TODO: I suggest to change request.Extension type to public Dictionary<string, object?>?
+            //request.Extensions ??= new();
+            //request.Extensions["persistedQuery"] = new Dictionary<string, object>
+            //{
+            //    ["version"] = APQ_SUPPORTED_VERSION,
+            //    ["sha256Hash"] = Hash.Compute(request.Query),
+            //};
+            request.Extensions ??= new
+            {
+                version = APQ_SUPPORTED_VERSION,
+                sha256Hash = Hash.Compute(request.Query),
+            };
+            request.Query = null;
+        }
+
+        var response = await SendHttpRequestAsync<TResponse>(request, cancellationToken);
+
+        if (useAPQ && response.Errors?.Length > 0)
+        {
+            if (response.Errors.Any(error => string.Equals(error.Message, "PersistedQueryNotFound", StringComparison.CurrentCultureIgnoreCase)))
+            {
+                // GraphQL server supports APQ!
+
+                // Alas, for the first time we did not guess and in vain removed Query, so we return Query and
+                // send request again. This is one-time "cache miss", not so scary. By the way, after a second call,
+                // there is a probability of receiving PERSISTED_QUERY_NOT_FOUND error again, but it is so ghostly
+                // that this can and should be neglect so not to complicate the code without the need.
```

Comment:
there is a small chance this would lead to an infinite loop, if we imagine that service is misconfigured and e.g. has issues with caching (or doesnt add new values due to lack of memory?). We prob should give up after several tries? But I agree it adds a lot of complexity and might be not worth it

Reply:
I think it's better to remove that comment at all.